### PR TITLE
Major update for worksheet header's look

### PR DESCRIPTION
**Pull request by Levilian**
1. standardized all the buttons and used grid for ordering
2. used material ui tooltip for keyboard shortcuts
3. enabled the toggling of web terminal from the header
4. combined worksheet information into one div

### CODE REVIEW

**Comment by Levilian** [comment]
Fix: #1456

**Comment by yipenghe** [comment]
Thanks for working on the title bar! I like the button designs and the hover tooltips. 

Would be great if you can include screenshots and walk us through the design. 

One thing I noticed is the worksheet section now spreads to full width, is this a side effect? Now it also weirdly has a slide horizontally in the ws area (see the slidebar at the very bottom)
Screenshot from 2019-10-16 00-28-55

I think it would be better if we can use the terminal toggle button to open/close the CLI (full height) directly. Currently after hiding the CLI, we need to 1) click show 2) click on CLI to open full height and to hide CLI we then need to 3) click outside the CLI to close it to let the hide button appear 4) click hide. Maybe we can show the CLI under the title bar, so the title bar section can stay fixed?

File: frontend/src/components/EditableField.js [hunk 1]
```diff
@@ -98,38 +98,6 @@ class EditableFieldBase extends React.Component<{
         );
     }
 
-    render() {
```
Comment (epicfaace) [inline frontend/src/components/EditableField.js:101]: haha I can't believe there were two render functions

File: frontend/src/components/worksheets/Worksheet.js [hunk 2]
```diff
@@ -806,40 +812,67 @@ class Worksheet extends React.Component {
 
         var searchClassName = this.state.showActionBar ? '': 'search-hidden';
         var editableClassName = canEdit ? 'editable' : '';
-        var viewClass = !canEdit && !this.state.editMode ? 'active' : '';
-        var rawClass = this.state.editMode ? 'active' : '';
         var disableWorksheetEditing = this.canEdit() ? '' : 'disabled';
-        var sourceStr = editPermission ? 'Edit Source' : 'View Source';
+        var sourceStr = editPermission ? 'Edit' : 'View';
```
Comment (epicfaace) [inline frontend/src/components/worksheets/Worksheet.js]: I think "Edit Source" is clearer because it makes it clear that we're actually editing the worksheet source.

Can you add screenshots to this PR of all new screens? (for example, how the "View" button looks like when you're actually editing the source)

File: frontend/src/components/worksheets/Worksheet.js [hunk 3]
```diff
@@ -806,40 +812,67 @@ class Worksheet extends React.Component {
...
-                    <button className={rawClass} onClick={this.editMode}>
-                        {sourceStr}
-                    </button>
-                    <button className={rawClass} onClick={e => this.delete()}>
-                        {"Delete"}
-                    </button>
-                </div>
-            </div>
+            <div onMouseMove={(ev) => {
+                ev.stopPropagation();
```
Comment (epicfaace) [inline frontend/src/components/worksheets/Worksheet.js]: Why did you add `ev.stopPropagation()`? It's not usually good to use it. https://css-tricks.com/dangers-stopping-event-propagation/

File: frontend/src/components/worksheets/Worksheet.js [hunk 4]
```diff
@@ -806,40 +812,67 @@ class Worksheet extends React.Component {
...
+                    Delete
+                </Button>
+            <Button
+                onClick={e => this.toggleActionBar()}
+                size='small'
+                color='inherit'
+                aria-label='Expand CLI'
+                >
+                    {this.state.showActionBar ? <ContractIcon className={classes.buttonIcon} />
+                : <ExpandIcon className={classes.buttonIcon} />}
```
Comment (epicfaace) [inline frontend/src/components/worksheets/Worksheet.js:849]: Also if you could add screenshots of this...

File: frontend/src/components/worksheets/Worksheet.js [hunk 5]
```diff
@@ -907,7 +940,7 @@ class Worksheet extends React.Component {
 
         return (
             <React.Fragment>
-                {action_bar_display}
+                                {action_bar_display}
```
Comment (epicfaace) [inline frontend/src/components/worksheets/Worksheet.js]:
```suggestion
                {action_bar_display}
```

File: frontend/src/components/worksheets/Worksheet.js [hunk 6]
```diff
@@ -949,7 +983,7 @@ const styles = (theme) => ({
         paddingBottom: 25,  // Height of Footer
     },
     worksheetOuter: {
-        maxWidth: 1200, // Worksheet width
+        maxWidth: 1960, // Worksheet width
```
Comment (epicfaace) [inline frontend/src/components/worksheets/Worksheet.js]: Yeah, as @yipenghe noticed, maybe it would be better to keep the width the same and change it in a PR.

File: frontend/src/components/worksheets/Worksheet.js [hunk 7]
```diff
@@ -923,6 +956,7 @@ class Worksheet extends React.Component {
                     onShowNewRun={() => this.setState({showNewRun: true})}
                     onShowNewText={() => this.setState({showNewText: true})}
                     />
+
```
Comment (epicfaace) [inline frontend/src/components/worksheets/Worksheet.js]: remove empty line

File: frontend/src/components/worksheets/WorksheetHeader.js [hunk 8]
```diff
@@ -5,116 +5,110 @@ import { WorksheetEditableField } from '../EditableField';
...
-    paddingTop: 10,
-    paddingBottom: 20,
+    paddingLeft: 28,
+    paddingRight: 28,
     boxShadow: "0px 2px 4px -2px rgba(0,0,0,0.2), 0px 4px 5px -1px rgba(0,0,0,0.14), 0px 1px 10px -1px rgba(0,0,0,0.12)"
 };
 
 export default ({ showActionBar, onShowNewUpload, onShowNewRun, onShowNewText, canEdit, info, classes, renderPermissions, reloadWorksheet, editButtons, anchorEl, setAnchorEl }) =>
-    <Sticky top={NAVBAR_HEIGHT - 6 + (showActionBar ? 60: 0)} innerZ={1059}>
+    <Sticky top={NAVBAR_HEIGHT - 6 + (showActionBar ? NAVBAR_HEIGHT : 0)} innerZ={1059}>
```
Comment (epicfaace) [inline frontend/src/components/worksheets/WorksheetHeader.js:21]: 👍

File: frontend/src/components/worksheets/WorksheetHeader.js [hunk 9]
```diff
@@ -5,116 +5,110 @@ import { WorksheetEditableField } from '../EditableField';
...
-                                onChange={() => reloadWorksheet()}
-                            />
-                        </h4>
-                    </Grid>
-                    <Grid item sm={12} md={5} container direction="column" justify="flex-end">
-                        <Grid item sm={12}>
-                            {info && <div className={classes.uuid}>{info.uuid}</div>}
+                <Grid container direction="column">
+                    <Grid container item xs={12} spacing={2} alignItems="flex-start" justify="space-between">
+                        <Grid item>
```
Comment (epicfaace) [inline frontend/src/components/worksheets/WorksheetHeader.js]: You can remove this `Grid` element

File: frontend/src/components/worksheets/WorksheetHeader.js [hunk 10]
```diff
@@ -5,116 +5,110 @@ import { WorksheetEditableField } from '../EditableField';
...
+                            <h5 className='worksheet-title'>
+                                {/*TODO: use html-content-editable*/}
```
Comment (epicfaace) [inline frontend/src/components/worksheets/WorksheetHeader.js]:
```suggestion
                                {/*TODO: use html contenteditable*/}
```

File: frontend/src/components/worksheets/WorksheetHeader.js [hunk 11]
```diff
@@ -5,116 +5,110 @@ import { WorksheetEditableField } from '../EditableField';
...
-                                                value={info && info.name}
-                                                uuid={info && info.uuid}
-                                                onChange={() => reloadWorksheet()}
-                                            />
-                                        </Grid>
-                                        <Grid item>
-                                            <span className={classes.label}>owner:</span>
+                        <Grid item style={{paddingTop: '10px'}}>
+                        {info && (
+                            <div>
```
Comment (epicfaace) [inline frontend/src/components/worksheets/WorksheetHeader.js]: You could use `React.Fragment` instead so it doesn't render an extra div tag

File: frontend/src/components/worksheets/items/ActionButtons.js [hunk 12]
```diff
@@ -19,41 +19,38 @@ class ActionButtons extends React.Component<{
                 onMouseMove={(ev) => {
                     ev.stopPropagation();
                 }}
-                className={classes.buttonsPanel}
+                // className={classes.buttonsPanel}
```
Comment (epicfaace) [inline frontend/src/components/worksheets/items/ActionButtons.js]: Can we remove this line?

File: frontend/src/components/worksheets/items/ActionButtons.js [hunk 13]
```diff
@@ -19,41 +19,38 @@ class ActionButtons extends React.Component<{
...
+                    <AddIcon className={classes.buttonIcon} />
+                    Cell
+                </Button>
+                <Button
+                    size='small'
+                    color='inherit'
                     aria-label='Add New Upload'
                     onClick={onShowNewUpload}
-                    classes={{ root: classes.buttonRoot }}
+                    // classes={{ root: classes.buttonRoot }}
```
Comment (epicfaace) [inline frontend/src/components/worksheets/items/ActionButtons.js]: remove these lines (and delete the corresponding values from the `classes` object) if we don't need them

File: frontend/src/components/worksheets/items/ActionButtons.js [hunk 14]
```diff
@@ -19,41 +19,38 @@ class ActionButtons extends React.Component<{
...
                 </Button>
                 <Button
-                    variant='outlined'
                     size='small'
-                    color='primary'
+                    color='inherit'
                     aria-label='Add New Run'
                     onClick={onShowNewRun}
-                    classes={{ root: classes.buttonRoot }}
+                    // startIcon={<RunIcon />}
```
Comment (epicfaace) [inline frontend/src/components/worksheets/items/ActionButtons.js]: rm line

File: frontend/src/css/imports.css [hunk 15]
```diff
@@ -7833,9 +7831,7 @@ fieldset[disabled] .worksheet_content .header-row .controls .edit-features .btn-
 }
 #command_line {
   text-align: left;
-  margin-left: auto;
   margin-right: auto;
-  max-width: 1700px;
```
Comment (epicfaace) [inline frontend/src/css/imports.css:7838]: why did you remove this line?

File: frontend/src/css/imports.css [hunk 16]
```diff
@@ -7833,9 +7831,7 @@ fieldset[disabled] .worksheet_content .header-row .controls .edit-features .btn-
 }
 #command_line {
   text-align: left;
-  margin-left: auto;
```
Comment (epicfaace) [inline frontend/src/css/imports.css:7836]: did this fix the centering issue with the web cli?

Comment (Levilian) [reply frontend/src/css/imports.css:7836]: yes

**Comment by Levilian** [comment]
## show terminal
Screen Shot 2019-10-16 at 11 50 21 AM

## hide terminal
Screen Shot 2019-10-16 at 11 49 32 AM

## edit source mode
Screen Shot 2019-10-16 at 11 49 44 AM

## shortcuts tooltip
Screen Shot 2019-10-16 at 11 50 05 AM

## access settings
Screen Shot 2019-10-16 at 11 50 12 AM

**Comment by Levilian** [comment]
> Thanks for working on the title bar! I like the button designs and the hover tooltips.
> 
> Would be great if you can include screenshots and walk us through the design.
> 
> One thing I noticed is the worksheet section now spreads to full width, is this a side effect? Now it also weirdly has a slide horizontally in the ws area (see the slidebar at the very bottom)
> Screenshot from 2019-10-16 00-28-55
> 
> I think it would be better if we can use the terminal toggle button to open/close the CLI (full height) directly. Currently after hiding the CLI, we need to 1) click show 2) click on CLI to open full height and to hide CLI we then need to 3) click outside the CLI to close it to let the hide button appear 4) click hide. Maybe we can show the CLI under the title bar, so the title bar section can stay fixed?

Thanks for the suggestion, Yipeng! 

To really change how the CLI looks, I think I would open a separate PR and think through the design choices. We've been keeping the CLI on top since the old version and I want to better understand the rationale for the old way before moving on to a new look.

File: frontend/src/components/worksheets/Worksheet.js (hunk 3 above)
Comment (epicfaace) [reply frontend/src/components/worksheets/Worksheet.js]: @Levilian
